tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -129,25 +107,58 @@ private async Task InitializeAsync()
                 Log.Error(e, "Initializing Live Debugger failed.");
             }
 
+            bool ShouldInitialize()
+            {
+                lock (GlobalLock)
+                {
+                    if (_isInitialized)
+                    {
+                        Log.Information("Live Debugger is already initialized.");
+                        return false;
+                    }
+
+                    if (!_settings.Enabled)
+                    {
+                        Log.Information("Live Debugger is disabled. To enable it, please set DD_DEBUGGER_ENABLED environment variable to 'true'.");
+                        return false;
+                    }
+
+                    if (_settings.TransportType != TracesTransportType.Default)
+                    {
+                        Log.Information("Live Debugger is not supported on UDS or named pipelines.");
+                        return false;
+                    }
+
+                    var isConfigurationSupported = !string.IsNullOrWhiteSpace(_discoveryService.ConfigurationEndpoint);
+                    if (!isConfigurationSupported)
+                    {
+                        Log.Warning("You must upgrade datadog-agent in order to leverage the Live Debugger. All debugging features will be disabled.");
+                        return false;
+                    }
+
+                    _isInitialized = true;
```

Comment:
Please move this out of the method - a method called `ShouldInitialize` should not have side effects or participate in the actual initialization. Perhaps we should also consider renaming it to something like `CheckCanInitialize`, since a verb method name seems more appropriate here, as we do have the side effect of writing logs either way.

Reply:
Agree, fixed at 10ea17673b64235f948ebde1f87ba0bee0454791